File: java/src/main/java/dominio/Cliente.java
package dominio;

import dominio.adapters.IAdapter;

public class Cliente {
    private IAdapter adapter;

    public void setAdapter(IAdapter adapter) {
        this.adapter = adapter;
    }

    public void ejecutar(){
        //TODO
        this.adapter.operacion();
        //TODO
    }
}

File: php/src/app/dominio/Cliente.php
<?php
namespace dominio;

use dominio\adapters\IAdapter;

class Cliente {
    /**
     * @var IAdapter
     */
    private $adapter;

    public function setAdapter(IAdapter $adapter){
        $this->adapter = $adapter;
    }

    public function ejecutar(){
        //TODO
        $this->adapter->operacion();
        //TODO
    }
}